cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake

```diff
@@ -767,6 +767,18 @@ else()
   # Search default search paths, after we search our own set of paths.
   cuda_find_host_program(CUDA_NVCC_EXECUTABLE nvcc)
 endif()
+
+# FAST_NVCC
+if(USE_FAST_NVCC AND CUDA_NVCC_EXECUTABLE AND NOT CUDA_NVCC_EXECUTABLE_ORIGIN)
+  set(CUDA_NVCC_EXECUTABLE_ORIGIN "${CUDA_NVCC_EXECUTABLE}")
+  set(FAST_NVCC_EXECUTABLE "${PROJECT_SOURCE_DIR}/tools/fast_nvcc/fast_nvcc.py")
+  configure_file(${PROJECT_SOURCE_DIR}/tools/fast_nvcc/wrap_nvcc.sh.in "${PROJECT_SOURCE_DIR}/tools/fast_nvcc/tmp/wrap_nvcc.sh")
+  file(COPY "${PROJECT_SOURCE_DIR}/tools/fast_nvcc/tmp/wrap_nvcc.sh"
+    DESTINATION "${PROJECT_SOURCE_DIR}/tools/fast_nvcc/"
+    FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
+  )
```

Comment:
Is this copy just to make sure the file has the correct permissions (as opposed to just writing directly to `"${PROJECT_SOURCE_DIR}/tools/fast_nvcc/"` from the `configure_file` line above)?

Reply:
correct.

Reply:
But from the docs, it looks like [`configure_file`](https://cmake.org/cmake/help/latest/command/configure_file.html) would propagate the permissions from `tools/fast_nvcc/wrap_nvcc.sh.in` by default, right (since `NO_SOURCE_PERMISSIONS` isn't passed)? So would it be sufficient to simply give the appropriate permissions to that file in Git, rather than adding them via `file` in CMake?

Reply:
Yeah it failed CI. Doesn't allow executable permission for .in files

Reply:
Ah I see :)